rnd/autogpt_builder/src/components/history.ts

```diff
@@ -0,0 +1,44 @@
+// history.ts
+
+type Action = {
+    type: 'ADD_NODE' | 'DELETE_NODE' | 'ADD_EDGE' | 'DELETE_EDGE' | 'UPDATE_NODE' | 'MOVE_NODE' | 'UPDATE_INPUT' | 'UPDATE_NODE_POSITION';
+    payload: any;
```

Comment:
The only other suggestion I would have here would be to avoid using `any` if at all possible.

```suggestion
type ActionPayload = NodePayload | EdgePayload | UpdateInputPayload | null;

```